libsolidity/analysis/TypeChecker.cpp

```diff
@@ -3391,6 +3392,8 @@ bool TypeChecker::visit(MemberAccess const& _memberAccess)
 			annotation.isPure = true;
 		else if (magicType->kind() == MagicType::Kind::MetaType && memberName == "interfaceId")
 			annotation.isPure = true;
+		else if (magicType->kind() == MagicType::Kind::MetaType && memberName == "typehash")
+			annotation.isPure = true;
```

Comment:
We need more validation on the struct's members here. See the bigger comment I left in the review.